aws/sdk/integration-tests/dynamodb/tests/shared-config.rs

```diff
@@ -35,6 +36,10 @@ async fn shared_config_testbed() {
 async fn service_config_from_profile() {
     let _ = tracing_subscriber::fmt::try_init();
 
+    // Remove env vars that could interfere with the test
+    env::remove_var("AWS_DEFAULT_REGION");
```

Comment:
Should we be more conservative and remove all `AWS_` environment variables?

Reply:
we should probably do that in the test runner at the codebuild level, but this only impacts this single test so being surgical seems like the more reasonable approach. Maybe we should do these unsets in https://github.com/smithy-lang/smithy-rs/blob/742aae9560230f1398ddbcb68ee0810d3af71828/tools/ci-scripts/check-aws-sdk-services#L16

Reply:
> but this only impacts this single test so being surgical seems like the more reasonable approach

Agreed I think this is more what I was suggesting is if we should consider this as a common test setup utility. Makes sense to unblock release at the moment.

Reply:
I think the correct long-term fix is to make it possible to replace the environment via the ConfigLoader. This should work for now.